mollie.php

```diff
@@ -1203,6 +1220,128 @@ public function getTabs()
+            'orderby' => false,
+            'search' => false,
+            'remove_onclick' => true,
+            'callback_object' => 'mollie',
+            'callback' => 'resendOrderPaymentLink'
+        ];
+    }
+
+    public function hookActionValidateOrder($params)
+    {
+        if($this->context->controller instanceof AdminOrdersControllerCore &&
+            $params["orderStatus"]->module_name === $this->name
+        ) {
+
+            $qrCode = false;
+            $paymentData = [
+                'amount' => [
+                    'value' => strval($params["order"]->total_paid),
+                    'currency' => $params["currency"]->iso_code,
+                ],
+                'redirectUrl' =>($qrCode
+                    ? $this->context->link->getModuleLink(
+                        'mollie',
+                        'qrcode',
+                        ['cart_id' => $params["cart"]->id, 'done' => 1, 'rand' => time()],
+                        true
+                    )
+                    : $this->context->link->getModuleLink(
+                        'mollie',
+                        'return',
+                        [
+                            'cart_id' => $params["cart"]->id,
+                            'utm_nooverride' => 1,
+                            'rand' => time(),
+                            'key' => $params["customer"]->secure_key,
+                            'customerId' => $params["customer"]->id
+                        ],
+                        true
+                    )
+                ),
+                'description' => $params["order"]->reference,
+                'metadata' => [
+                    'cart_id' => $params["order"]->reference,
+                    'order_reference' => $params["order"]->reference,
+                    'secure_key' => $params["order"]->secure_key
+                ],
+            ];
+
+            if (!Mollie\Utility\EnvironmentUtility::isLocalEnvironment()) {
+                $paymentData['webhookUrl'] = $this->context->link->getModuleLink(
+                    'mollie',
+                    'webhook',
+                    [],
+                    true
+                );
+            }
+
+            $newPayment = $this->api->payments->create($paymentData);
+
+            Db::getInstance()->insert(
+                'mollie_payments',
+                [
+                    'cart_id'        => (int) $params["cart"]->id,
+                    'method'         => pSQL($params["order"]->payment),
+                    'transaction_id' => $newPayment->id,
+                    'bank_status'    => _PhpScoper5eddef0da618a\Mollie\Api\Types\PaymentStatus::STATUS_OPEN,
+                    'order_id'       => (int) $params["order"]->id,
+                    'order_reference'=> (int) $params["order"]->reference,
+                    'created_at'     => array('type' => 'sql', 'value' => 'NOW()'),
+                ]
+            );
+        }
+        return $params;
+    }
```

Comment:
move this spaggetti to some service and split logic a bit - I dont have a clue what you are doing here